test/inspector/test-bindings.js

```diff
@@ -0,0 +1,111 @@
+'use strict';
+require('../common');
+const assert = require('assert');
+const path = require('path');
+
+// This test case will set a breakpoint 4 lines below
+function debuggedFunction() {
+  let i;
+  let accum = 0;
+  for (i = 0; i < 5; i++) {
+    accum += i;
+  }
+  return accum;
+}
+
+let scopeCallback = null;
+let id = 1;
+const callbacks = new Map();
+
+function checkScope(session, scopeId) {
+  assert.ok(scopeCallback);
+  sendMessage(session, 'Runtime.getProperties', {
+    'objectId': scopeId,
+    'ownProperties': false,
+    'accessorPropertiesOnly': false,
+    'generatePreview': true
+  }, scopeCallback);
+}
+
+function checkResponse(result, error) {
+  assert.ok(result);
+  assert.ok(!error);
+}
+
+function inspectorMessageCallback(session, notication) {
+  const message = JSON.parse(notication);
+  const id = message['id'];
+  if (id) {
+    callbacks.get(id)(message['result'], message['error']);
+    callbacks.delete(id);
+  } else if (message['method'] == 'Debugger.paused') {
+    const params = message['params'];
+    const callFrame = params['callFrames'][0];
+    const scopeId = callFrame['scopeChain'][0]['object']['objectId'];
+    checkScope(session, scopeId);
+  }
+}
+
+function sendMessage(session, method, params, callback) {
+  const message_id = id++;
+  const message = {id: message_id, method: method};
+  if (params) {
+    message['params'] = params;
+  }
+  callbacks.set(message_id, callback || checkResponse);
+  session.dispatch(JSON.stringify(message));
+}
+
+function testExecution() {
+  let cur = 0;
+  const failures = [];
+  const expects = {
+    i: [0, 1, 2, 3, 4],
+    accum: [0, 0, 1, 3, 6]
+  };
+  scopeCallback = function(result) {
+    const i = cur++;
+    let v, actual, expected;
+    for (v of result['result']) {
+      actual = v['value']['value'];
+      expected = expects[v['name']][i];
+      if (actual != expected) {
+        failures.push('Iteration ' + i + ' variable: ' + v['name'] +
+                      ' expected: ' + expected + ' actual: ' + actual);
+      }
+    }
+  };
+  const breakpoint = {
+    'lineNumber': 10,
+    'url': path.resolve(__dirname, __filename),
+    'columnNumber': 0,
+    'condition': ''
+  };
+  const session = process.inspector.connect(
+      (notification) => inspectorMessageCallback(session, notification));
+  sendMessage(session, 'Debugger.enable');
+  sendMessage(session, 'Debugger.setBreakpointByUrl', breakpoint);
+
+  debuggedFunction();
+
+  assert.deepStrictEqual(failures, []);
+  assert.strictEqual(cur, 5);
+  scopeCallback = null;
+  session.disconnect();
```

Comment:
I'd also add test for:
- connect twice;
- disconnect twice;
- disconnect without connect;
- send 1000 messages in a row;
- pass non-function as callback;
- dispatch without connect.